### ISSUE TITLE
`napari-deeplabcut` cannot save annotated h5 file due to `KeyError: 'root'`

### ISSUE THREAD

**Issue by YugaKimura** [DeepLabCut/DeepLabCut#2752]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Bug description

When I annotate an image and try to save the layer using Ctrl+S in the GUI, I get an error keyeroor:root at metadata=["root"] in _writer.py. viewer.layers[1].save("annotated.csv"), the csv file comes up, but viewer.layers[1].save("annotated.h5") also gives the above error. How can I deal with this error and save the annotated data?

### Operating System

Ubuntu 22.04.3 LTS

### DeepLabCut version

2.3.9(Version to 2.3.10 doesn't solve the problem.)

### DeepLabCut mode

single animal

### Device type

gpu
```
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.146.02             Driver Version: 535.146.02   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 2080 Ti     Off | 00000000:01:00.0  On |                  N/A |
| 27%   35C    P8              16W / 250W |    713MiB / 11264MiB |      3%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A      1442      G   /usr/lib/xorg/Xorg                           35MiB |
|    0   N/A  N/A      4683      G   /usr/lib/xorg/Xorg                           35MiB |
|    0   N/A  N/A      5611      G   /usr/lib/xorg/Xorg                          192MiB |
|    0   N/A  N/A      5734      G   /usr/bin/gnome-shell                         77MiB |
|    0   N/A  N/A    695090      G   ...sion,SpareRendererForSitePerProcess       46MiB |
|    0   N/A  N/A   3220642      G   firefox                                     153MiB |
|    0   N/A  N/A   3230162      G   ...pyenv/versions/kimuradlc/bin/python       15MiB |
+---------------------------------------------------------------------------------------+
```

### Steps To Reproduce

1. annotate 25 .png images
2. save layer
3. occur error

### Relevant log output

```shell
Traceback (most recent call last):
  File "/home/erato_0/.pyenv/versions/kimuradlc/lib/python3.10/site-packages/napari_deeplabcut/_widgets.py", line 656, in <lambda>
    lambda: _save_layers_dialog(
  File "/home/erato_0/.pyenv/versions/kimuradlc/lib/python3.10/site-packages/napari_deeplabcut/_widgets.py", line 300, in _save_layers_dialog
    self.viewer.layers.save("", selected=True, plugin="napari-deeplabcut")
  File "/home/erato_0/.pyenv/versions/kimuradlc/lib/python3.10/site-packages/napari/utils/_proxies.py", line 188, in __call__
    return self.create(self.__wrapped__(*args, **kwargs))
  File "/home/erato_0/.pyenv/versions/kimuradlc/lib/python3.10/site-packages/napari/components/layerlist.py", line 476, in save
    return save_layers(path, layers, plugin=plugin, _writer=_writer)
  File "/home/erato_0/.pyenv/versions/kimuradlc/lib/python3.10/site-packages/napari/plugins/io.py", line 229, in save_layers
    _written, writer_name = _write_single_layer_with_plugins(
  File "/home/erato_0/.pyenv/versions/kimuradlc/lib/python3.10/site-packages/napari/plugins/io.py", line 442, in _write_single_layer_with_plugins
    written_paths, writer_name = _npe2.write_layers(
  File "/home/erato_0/.pyenv/versions/kimuradlc/lib/python3.10/site-packages/napari/plugins/_npe2.py", line 111, in write_layers
    paths, writer = io_utils.write_get_writer(
  File "/home/erato_0/.pyenv/versions/kimuradlc/lib/python3.10/site-packages/npe2/io_utils.py", line 115, in write_get_writer
    return _write(path, layer_data, plugin_name=plugin_name, return_writer=True)
  File "/home/erato_0/.pyenv/versions/kimuradlc/lib/python3.10/site-packages/npe2/io_utils.py", line 320, in _write
    res = writer.exec(args=args)
  File "/home/erato_0/.pyenv/versions/kimuradlc/lib/python3.10/site-packages/npe2/manifest/utils.py", line 61, in exec
    return self.get_callable(reg)(*args, **kwargs)
  File "/home/erato_0/.pyenv/versions/kimuradlc/lib/python3.10/site-packages/napari_deeplabcut/_writer.py", line 51, in write_hdf
    root = metadata["root"]
KeyError: 'root'
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by n-poulsen** [comment]
Hi @YugaKimura! Could you please let me know what the output of the `napari --info` command is in your environment?

**Comment by j-diefenbach** [comment]
I'm having the same issue, after updating to latest deeplabcut version still. Happens when annotating 1 frame or more
napari --info output (excluding full settings path):
```
napari: 0.4.18
Platform: Windows-10-10.0.19045-SP0
Python: 3.10.14 | packaged by conda-forge | (main, Mar 20 2024, 12:40:08) [MSC v.1938 64 bit (AMD64)]
Qt: 6.4.2
PySide6:
NumPy: 1.26.4
SciPy: 1.10.1
Dask: 2024.9.1
VisPy: 0.12.2
magicgui: 0.9.1
superqt: 0.6.7
in-n-out: 0.2.1
app-model: 0.2.8
npe2: 0.7.7

OpenGL:
  - GL version:  4.6.0 NVIDIA 560.76
  - MAX_TEXTURE_SIZE: 32768

Screens:
  - screen 1: resolution 1800x1131, scale 1.0
Settings path: ... settings.yaml
Plugins:
  - napari: 0.4.18 (77 contributions)
  - napari-console: 0.1.0 (0 contributions)
  - napari-deeplabcut: 0.2.0 (16 contributions)
  - napari-svg: 0.2.0 (2 contributions)
```

**Comment by n-poulsen** [comment]
The issue seems to be with the `napari-deeplabcut: 0.2.0` version installed. Could you run `pip install napari-deeplabcut==0.2.1.7` in your conda environment and let me know if the error persists?

**Comment by j-diefenbach** [comment]
> The issue seems to be with the `napari-deeplabcut: 0.2.0` version installed. Could you run `pip install napari-deeplabcut==0.2.1.7` in your conda environment and let me know if the error persists?

Still persists, 0.2.1.7 was the version I was using before I tried to see if 0.2.0 was a fix. Changing back to 0.2.1.7 doesn’t change anything for the error message

**Comment by j-diefenbach** [comment]
If it is relevant deeplabcut was installed via pip for my environment

**Comment by n-poulsen** [comment]
Thanks for the info - could you let me know all package versions that are installed in your conda environment (the output of `pip list`)?

**Comment by j-diefenbach** [comment]
> Thanks for the info - could you let me know all package versions that are installed in your conda environment (the output of `pip list`)?

I created a fresh environment, installed and followed exactly from https://deeplabcut.github.io/DeepLabCut/docs/installation.html and was still able to replicate the error (same key error for meta['root']).
Update: Also tried downgrading to 3.0.0rc1 but have the same issue. Anything earlier requires a bunch of tensorflow packages that aren't installed w the environment so didn't try those
Pip list output (for fresh env, although might have some other packages) is:
```alabaster                     0.7.16
albumentations                1.4.3
anyio                         4.6.2.post1
app-model                     0.2.8
appdirs                       1.4.4
argon2-cffi                   23.1.0
argon2-cffi-bindings          21.2.0
arrow                         1.3.0
asttokens                     2.4.1
async-lru                     2.0.4
attrs                         24.2.0
babel                         2.16.0
beautifulsoup4                4.12.3
bleach                        6.1.0
blosc2                        2.0.0
Brotli                        1.1.0
build                         1.2.2.post1
cached-property               1.5.2
cachey                        0.2.1
certifi                       2024.8.30
cffi                          1.17.1
charset-normalizer            3.4.0
click                         8.1.7
cloudpickle                   3.1.0
colorama                      0.4.6
comm                          0.2.2
contourpy                     1.3.0
cycler                        0.12.1
Cython                        3.0.11
dask                          2024.9.1
dask-expr                     1.1.15
dask-image                    2024.5.3
debugpy                       1.8.7
decorator                     5.1.1
deeplabcut                    3.0.0rc5
defusedxml                    0.7.1
dlclibrary                    0.0.7
docker-pycreds                0.4.0
docstring_parser              0.16
docutils                      0.17.1
einops                        0.8.0
entrypoints                   0.4
exceptiongroup                1.2.2
executing                     2.1.0
fastjsonschema                2.20.0
filelock                      3.16.1
filterpy                      1.4.5
flexcache                     0.3
flexparser                    0.3.1
fonttools                     4.54.1
fqdn                          1.5.1
freetype-py                   2.5.1
fsspec                        2024.9.0
gitdb                         4.0.11
GitPython                     3.1.43
h11                           0.14.0
h2                            4.1.0
HeapDict                      1.0.1
hpack                         4.0.0
hsluv                         5.0.4
httpcore                      1.0.6
httpx                         0.27.2
huggingface-hub               0.26.1
hyperframe                    6.0.1
idna                          3.10
imageio                       2.36.0
imageio-ffmpeg                0.5.1
imagesize                     1.4.1
imgaug                        0.4.0
importlib_metadata            8.5.0
importlib_resources           6.4.5
in-n-out                      0.2.1
ipykernel                     6.29.5
ipython                       8.28.0
ipython_genutils              0.2.0
ipywidgets                    8.1.5
isoduration                   20.11.0
jedi                          0.19.1
Jinja2                        3.1.4
joblib                        1.4.2
json5                         0.9.25
jsonpointer                   3.0.0
jsonschema                    4.23.0
jsonschema-specifications     2024.10.1
jupyter                       1.1.1
jupyter_client                8.6.3
jupyter-console               6.6.3
jupyter_core                  5.7.2
jupyter-events                0.10.0
jupyter-lsp                   2.2.5
jupyter_server                2.14.2
jupyter_server_terminals      0.5.3
jupyterlab                    4.2.5
jupyterlab_pygments           0.3.0
jupyterlab_server             2.27.3
jupyterlab_widgets            3.0.13
kiwisolver                    1.4.7
lazy_loader                   0.4
llvmlite                      0.43.0
locket                        1.0.0
magicgui                      0.9.1
markdown-it-py                3.0.0
MarkupSafe                    3.0.1
matplotlib                    3.8.4
matplotlib-inline             0.1.7
mdurl                         0.1.2
mistune                       3.0.2
mpmath                        1.3.0
msgpack                       1.1.0
napari                        0.4.18
napari-console                0.1.0
napari-deeplabcut             0.2.1.7
napari-plugin-engine          0.2.0
napari-svg                    0.2.0
natsort                       8.4.0
nb_conda                      2.2.1
nb_conda_kernels              2.5.1
nbclassic                     1.1.0
nbclient                      0.10.0
nbconvert                     7.16.4
nbformat                      5.10.4
ndindex                       1.9.2
nest-asyncio                  1.6.0
networkx                      3.4.1
notebook                      6.5.7
notebook_shim                 0.2.4
npe2                          0.7.7
numba                         0.60.0
numexpr                       2.10.1
numpy                         1.26.4
numpydoc                      1.5.0
opencv-python                 4.10.0.84
opencv-python-headless        4.10.0.84
overrides                     7.7.0
packaging                     24.1
pandas                        2.2.3
pandocfilters                 1.5.0
parso                         0.8.4
partd                         1.4.2
patsy                         0.5.6
pickleshare                   0.7.5
pillow                        10.4.0
PIMS                          0.7
Pint                          0.24.3
pip                           24.2
pkgutil_resolve_name          1.3.10
platformdirs                  4.3.6
pooch                         1.8.2
prometheus_client             0.21.0
prompt_toolkit                3.0.48
protobuf                      5.28.2
psutil                        6.0.0
psygnal                       0.11.1
pure_eval                     0.2.3
py-cpuinfo                    9.0.0
pyarrow                       17.0.0
pycocotools                   2.0.8
pyconify                      0.1.6
pycparser                     2.22
pydantic                      1.10.18
pydantic-compat               0.1.2
Pygments                      2.18.0
PyOpenGL                      3.1.7
pyparsing                     3.2.0
pyproject_hooks               1.2.0
PySide6                       6.4.2
PySide6-Addons                6.4.2
PySide6-Essentials            6.4.2
PySocks                       1.7.1
python-dateutil               2.9.0.post0
python-json-logger            2.0.7
pytz                          2024.2
pywin32                       308
pywinpty                      2.0.14
PyYAML                        6.0.2
pyzmq                         26.2.0
QDarkStyle                    3.1
qtconsole                     5.6.0
QtPy                          2.4.1
referencing                   0.35.1
requests                      2.32.3
rfc3339-validator             0.1.4
rfc3986-validator             0.1.1
rich                          13.9.2
rpds-py                       0.20.0
ruamel.yaml                   0.18.6
ruamel.yaml.clib              0.2.12
safetensors                   0.4.5
scikit-image                  0.24.0
scikit-learn                  1.5.2
scipy                         1.10.1
Send2Trash                    1.8.3
sentry-sdk                    2.17.0
setproctitle                  1.3.3
setuptools                    75.1.0
shapely                       2.0.6
shellingham                   1.5.4
shiboken6                     6.4.2
six                           1.16.0
slicerator                    1.1.0
smmap                         5.0.1
sniffio                       1.3.1
snowballstemmer               2.2.0
soupsieve                     2.5
Sphinx                        4.5.0
sphinxcontrib-applehelp       2.0.0
sphinxcontrib-devhelp         2.0.0
sphinxcontrib-htmlhelp        2.1.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          2.0.0
sphinxcontrib-serializinghtml 2.0.0
stack-data                    0.6.3
statsmodels                   0.14.4
superqt                       0.6.7
sympy                         1.13.1
tables                        3.8.0
terminado                     0.18.1
threadpoolctl                 3.5.0
tifffile                      2024.9.20
timm                          1.0.11
tinycss2                      1.3.0
tomli                         2.0.2
tomli_w                       1.1.0
toolz                         1.0.0
torch                         2.5.0
torchvision                   0.20.0
tornado                       6.4.1
tqdm                          4.66.5
traitlets                     5.14.3
typer                         0.12.5
types-python-dateutil         2.9.0.20241003
typing_extensions             4.12.2
typing-utils                  0.1.0
tzdata                        2024.2
uri-template                  1.3.0
urllib3                       2.2.3
vispy                         0.12.2
wandb                         0.18.5
wcwidth                       0.2.13
webcolors                     24.8.0
webencodings                  0.5.1
websocket-client              1.8.0
wheel                         0.44.0
widgetsnbextension            4.0.13
win_inet_pton                 1.1.0
wrapt                         1.16.0
zipp                          3.20.2
zstandard                     0.23.0
```

**Comment by j-diefenbach** [comment]
Not sure if intended or not, but when selecting "check labels" or "add labels" the napari window is empty. This might be related since I have to manually add them via the "add with plugin". 

I created another project and repeated the process from scratch this time in a different folder by accident. This time the add labels menu auto-loaded the corect files and setup. The h5 files and labels seem to have saved perfectly fine? Not sure if it was something to do with the location / setup of files with the given version or if it was due to a different file location, I'll check and add a comment here if its clear what the issue had been for others having the same issue.

In summary looks like a weird issue on my end, probably a similar issue for @YugaKimura. It may have been that the config file was old/corrupted for the projects I had tested.

**Comment by n-poulsen** [comment]
Thanks for coming back to us with this @j-diefenbach - I think you're likely correct and the issue comes from a corrupt configuration file. Using the "add with plugin" feature is likely because napari failed to load the `napari-deeplabcut` plugin with that project, which could lead to the `root` error.

I'll close this issue as I can't reproduce it, and if you're able to figure out why it's failing to use `napari-deeplabcut` with your project please re-open it.
